x-pack/legacy/plugins/apm/public/components/app/TraceLink/index.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import url from 'url';
+import styled from 'styled-components';
+import { Redirect } from 'react-router-dom';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { QueryParams } from '../../shared/Links/apm/ExternalLinks';
+
+const CentralizedContainer = styled.div`
+  height: 100%;
+  display: flex;
+`;
+
+const redirectToTransactionDetailPage = ({
+  transaction,
+  rangeFrom,
+  rangeTo
+}: { transaction: Transaction } & Partial<QueryParams>) => {
+  const dateRange = rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {};
+  return url.format({
+    pathname: `/services/${transaction.service.name}/transactions/view`,
+    query: {
+      traceId: transaction.trace.id,
+      transactionId: transaction.transaction.id,
+      transactionName: transaction.transaction.name,
+      transactionType: transaction.transaction.type,
+      ...dateRange
+    }
+  });
+};
+
+const redirectToTracePage = (traceId: string) =>
+  url.format({
+    pathname: `/traces`,
+    query: {
+      kuery: encodeURIComponent(`trace.id : "${traceId}"`)
+    }
+  });
+
+export const TraceLink = () => {
+  const { urlParams } = useUrlParams();
+  const { traceIdLink: traceId, rangeFrom, rangeTo } = urlParams;
+
+  const { data = { transaction: null }, status } = useFetcher(
+    callApmApi => {
+      if (traceId) {
+        return callApmApi({
+          pathname: '/api/apm/transaction/{traceId}',
+          params: {
+            path: {
+              traceId
+            }
+          }
+        });
+      }
+    },
+    [traceId]
+  );
+  if (traceId && status === FETCH_STATUS.SUCCESS) {
+    const to = data.transaction
+      ? redirectToTransactionDetailPage({
+          transaction: data.transaction,
+          rangeFrom,
+          rangeTo
+        })
+      : redirectToTracePage(traceId);
```

Comment:
What about the date range for the trace page?

Reply:
I was wondering if that is really necessary as if the trace page is opened it means that no transaction was found.